### PR TITLE
liveview: Config based error handling with OnViewError

### PR DESCRIPTION
**Pull request by floodfx**

### CODE REVIEW

File: live/liveview.go [hunk 1]
```diff
@@ -36,6 +35,12 @@ type Config struct {
 	//  - Pass the LayoutDot to the liveViewContainerTag (i.e. {{ liveViewContainerTag .LayoutDot }})
 	//  - Set the CSRF token in a meta tag (i.e. <meta name="csrf-token" content="{{ .LayoutDot.CSRFToken }}">)
 	RenderLayout func(http.ResponseWriter, *http.Request, *LayoutDot) (any, *htmltmpl.Template)
+	// OnViewError is called when an error occurs during a View lifecycle method (e.g. HandleEvent, HandleInfo, etc)
+	// AND the view is connected to a socket (as opposed to the initial HTTP request). If nil, the error is not logged.
```
Comment (josharian) [inline live/liveview.go:39]: I don't understand the sentence starting "If nil...". Can we delete it?

File: live/liveview.go [hunk 2]
```diff
@@ -36,6 +35,12 @@ type Config struct {
...
+	// In any case, the javascript client will receive a "phx_error" message via the connected socket which in the case
```
Comment (josharian) [inline live/liveview.go:40]: s/In any case/Regardless of whether OnViewError is non-nil/

File: live/liveview.go [hunk 3]
```diff
@@ -276,29 +281,41 @@ func (s *socket) serve(ctx context.Context) {
 		select {
 		case info := <-s.info:
 			r, err = s.handleInfo(ctx, info)
-			res = append(res, r)
+			if r != nil {
```
Comment (josharian) [inline live/liveview.go:284]: `if err == nil` would be a safer check. same for similar lines below.

File: live/liveview.go (hunk 1 above)
Comment (josharian) [reply live/liveview.go:39]: Oh, I see, the nil thing is the OnViewError. I'd say: "OnViewError may be nil."

**Review by josharian — APPROVED** [review]

File: live/liveview.go (hunk 3 above)
Comment (floodfx) [reply live/liveview.go:284]: went back and forth here.  originally checked error :)

**Pull request merged**